app/src/main/java/at/bitfire/icsdroid/model/SubscriptionSettingsModel.kt

```diff
@@ -1,38 +1,45 @@
 package at.bitfire.icsdroid.model
 
 import android.net.Uri
-import androidx.lifecycle.MediatorLiveData
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import at.bitfire.icsdroid.HttpUtils
 import at.bitfire.icsdroid.db.entity.Subscription
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import java.net.URISyntaxException
 
 class SubscriptionSettingsModel : ViewModel() {
-    val url = MutableLiveData<String?>(null)
-    val fileName = MutableLiveData<String?>(null)
-    val urlError = MutableLiveData<String?>(null)
-    val title = MutableLiveData<String?>(null)
-    val color = MutableLiveData<Int?>(null)
-    val ignoreAlerts = MutableLiveData(false)
-    val defaultAlarmMinutes = MutableLiveData<Long?>(null)
-    val defaultAllDayAlarmMinutes = MutableLiveData<Long?>(null)
+    val url = MutableStateFlow<String?>(null)
+    val fileName = MutableStateFlow<String?>(null)
+    val urlError = MutableStateFlow<String?>(null)
+    val title = MutableStateFlow<String?>(null)
+    val color = MutableStateFlow<Int?>(null)
+    val ignoreAlerts = MutableStateFlow(false)
+    val defaultAlarmMinutes = MutableStateFlow<Long?>(null)
+    val defaultAllDayAlarmMinutes = MutableStateFlow<Long?>(null)
 
-    // advanced
-    val ignoreDescription = MutableLiveData(false)
+    // advanced settings
+    val ignoreDescription = MutableStateFlow(false)
```

Comment:
Same, please, replace all `MutableStateFlow` by Compose states.

Reply:
Before using flows we were merging this as live data. Now we "combine" it as flows in `EditCalendarActivity`. I would prefer not to use compose `derivedStateOf` for the matter of simplicity.

Reply:
Sounds reasonable